File: RAT/Hades v1.6/Source/xRAT.vb/xRAT/PortScanner.vb
Imports Microsoft.VisualBasic
Imports Microsoft.VisualBasic.CompilerServices
Imports System
Imports System.ComponentModel
Imports System.Diagnostics
Imports System.Drawing
Imports System.Net
Imports System.Net.Sockets
Imports System.Runtime.CompilerServices
Imports System.Threading
Imports System.Windows.Forms

Namespace xRAT
    <DesignerGenerated> _
    Public Class PortScanner
        Inherits Form
        ' Methods
        Public Sub New()
            Class1.QaIGh5M7cuigS
            Me.InitializeComponent
        End Sub

        Private Sub Button1_Click(ByVal sender As Object, ByVal e As EventArgs)
            If Me.bool_0 Then
                Me.method_2(True)
            Else
                Me.method_2(False)
                Me.ListBox1.Items.Clear
                Me.ProgressBar1.Maximum = Convert.ToInt32(Decimal.Add(Me.NumericUpDown2.Value, Decimal.One))
                Me.ProgressBar1.Minimum = Convert.ToInt32(Me.NumericUpDown1.Value)
                Me.ProgressBar1.Value = Convert.ToInt32(Me.NumericUpDown1.Value)
                Try 
                    Dim address As IPAddress = Me.method_0(Me.TextBox1.Text)
                    Dim num As Integer = 0
                    Do
                        New Thread(New ParameterizedThreadStart(AddressOf Me.method_3)).Start(New Object() { Decimal.Add(Me.NumericUpDown1.Value, New Decimal(num)), Me.NumericUpDown2.Value, 8, address, Me.NumericUpDown3.Value })
                        num += 1
                    Loop While (num <= 7)
                Catch exception1 As Exception
                    ProjectData.SetProjectError(exception1)
                    ProjectData.ClearProjectError
                End Try
            End If
        End Sub

        <DebuggerNonUserCode> _
        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            Try 
                If (disposing AndAlso (Not Me.icontainer_0 Is Nothing)) Then
                    Me.icontainer_0.Dispose
                End If
            Finally
                MyBase.Dispose(disposing)
            End Try
        End Sub

        <DebuggerStepThrough> _
        Private Sub InitializeComponent()
            Dim manager As New ComponentResourceManager(GetType(PortScanner))
            Me.NumericUpDown1 = New NumericUpDown
            Me.NumericUpDown2 = New NumericUpDown
            Me.TextBox1 = New TextBox
            Me.Label2 = New Label
            Me.Label3 = New Label
            Me.ListBox1 = New ListBox
            Me.NumericUpDown3 = New NumericUpDown
            Me.Label4 = New Label
            Me.Button1 = New Button
            Me.Label5 = New Label
            Me.ProgressBar1 = New ProgressBar
            Me.NumericUpDown1.BeginInit
            Me.NumericUpDown2.BeginInit
            Me.NumericUpDown3.BeginInit
            Me.SuspendLayout
            Me.NumericUpDown1.Font = New Font("Verdana", 8.25!, FontStyle.Regular, GraphicsUnit.Point, 0)
            Dim point As New Point(&H80, 70)
            Me.NumericUpDown1.Location = point
            Dim num As New Decimal(New Integer() { &HFFFF, 0, 0, 0 })
            Me.NumericUpDown1.Maximum = num
            Me.NumericUpDown1.Name = "NumericUpDown1"
            Dim size As New Size(60, &H15)
            Me.NumericUpDown1.Size = size
            Me.NumericUpDown1.TabIndex = 2
            Me.NumericUpDown1.TextAlign = HorizontalAlignment.Center
            Me.NumericUpDown2.Font = New Font("Verdana", 8.25!, FontStyle.Regular, GraphicsUnit.Point, 0)
            point = New Point(&HC2, 70)
            Me.NumericUpDown2.Location = point
            num = New Decimal(New Integer() { &HFFFF, 0, 0, 0 })
            Me.NumericUpDown2.Maximum = num
            Me.NumericUpDown2.Name = "NumericUpDown2"
            size = New Size(60, &H15)
            Me.NumericUpDown2.Size = size
            Me.NumericUpDown2.TabIndex = 3
            Me.NumericUpDown2.TextAlign = HorizontalAlignment.Center
            Me.TextBox1.Font = New Font("Verdana", 8.25!, FontStyle.Regular, GraphicsUnit.Point, 0)
            point = New Point(&H80, &H19)
            Me.TextBox1.Location = point
            Me.TextBox1.Name = "TextBox1"
            size = New Size(&H7E, &H15)
            Me.TextBox1.Size = size
            Me.TextBox1.TabIndex = 4
            Me.TextBox1.Text = "127.0.0.1"
            Me.TextBox1.TextAlign = HorizontalAlignment.Center
            Me.Label2.AutoSize = True
            Me.Label2.Font = New Font("Verdana", 8.25!, FontStyle.Regular, GraphicsUnit.Point, 0)
            point = New Point(&H7D, &H36)
            Me.Label2.Location = point
            Dim padding As New Padding(3, 5, 3, 0)
            Me.Label2.Margin = padding
            Me.Label2.Name = "Label2"
            size = New Size(70, 13)
            Me.Label2.Size = size
            Me.Label2.TabIndex = 5
            Me.Label2.Text = "Port Range"
            Me.Label3.AutoSize = True
            Me.Label3.Font = New Font("Verdana", 8.25!, FontStyle.Regular, GraphicsUnit.Point, 0)
            point = New Point(&H7D, 9)
            Me.Label3.Location = point
            Me.Label3.Name = "Label3"
            size = New Size(&H2C, 13)
            Me.Label3.Size = size
            Me.Label3.TabIndex = 7
            Me.Label3.Text = "Target"
            Me.ListBox1.FormattingEnabled = True
            point = New Point(15, &H19)
            Me.ListBox1.Location = point
            Me.ListBox1.Name = "ListBox1"
            size = New Size(&H6B, &HBA)
            Me.ListBox1.Size = size
            Me.ListBox1.TabIndex = 8
            Me.NumericUpDown3.Font = New Font("Verdana", 8.25!, FontStyle.Regular, GraphicsUnit.Point, 0)
            point = New Point(&H80, &H73)
            Me.NumericUpDown3.Location = point
            num = New Decimal(New Integer() { &HEA60, 0, 0, 0 })
            Me.NumericUpDown3.Maximum = num
            Me.NumericUpDown3.Name = "NumericUpDown3"
            size = New Size(&H7E, &H15)
            Me.NumericUpDown3.Size = size
            Me.NumericUpDown3.TabIndex = 9
            Me.NumericUpDown3.TextAlign = HorizontalAlignment.Center
            num = New Decimal(New Integer() { 800, 0, 0, 0 })
            Me.NumericUpDown3.Value = num
            Me.Label4.AutoSize = True
            Me.Label4.Font = New Font("Verdana", 8.25!, FontStyle.Regular, GraphicsUnit.Point, 0)
            point = New Point(&H81, &H63)
            Me.Label4.Location = point
            padding = New Padding(3, 5, 3, 0)
            Me.Label4.Margin = padding
            Me.Label4.Name = "Label4"
            size = New Size(&H35, 13)
            Me.Label4.Size = size
            Me.Label4.TabIndex = 10
            Me.Label4.Text = "Timeout"
            Me.Button1.Font = New Font("Verdana", 8.25!, FontStyle.Regular, GraphicsUnit.Point, 0)
            point = New Point(&H80, 180)
            Me.Button1.Location = point
            Me.Button1.Name = "Button1"
            size = New Size(&H7E, &H1F)
            Me.Button1.Size = size
            Me.Button1.TabIndex = 11
            Me.Button1.Text = "Scan Ports"
            Me.Button1.UseVisualStyleBackColor = True
            Me.Label5.AutoSize = True
            Me.Label5.Font = New Font("Verdana", 8.25!, FontStyle.Regular, GraphicsUnit.Point, 0)
            point = New Point(12, 9)
            Me.Label5.Location = point
            Me.Label5.Name = "Label5"
            size = New Size(70, 13)
            Me.Label5.Size = size
            Me.Label5.TabIndex = 12
            Me.Label5.Text = "Open Ports"
            point = New Point(&H80, &H97)
            Me.ProgressBar1.Location = point
            Me.ProgressBar1.Name = "ProgressBar1"
            size = New Size(&H7E, &H17)
            Me.ProgressBar1.Size = size
            Me.ProgressBar1.TabIndex = 13
            Dim ef As New SizeF(6!, 13!)
            Me.AutoScaleDimensions = ef
            Me.AutoScaleMode = AutoScaleMode.Font
            size = New Size(&H108, &HDD)
            Me.ClientSize = size
            Me.Controls.Add(Me.ProgressBar1)
            Me.Controls.Add(Me.Label5)
            Me.Controls.Add(Me.Button1)
            Me.Controls.Add(Me.Label4)
            Me.Controls.Add(Me.NumericUpDown3)
            Me.Controls.Add(Me.ListBox1)
            Me.Controls.Add(Me.Label3)
            Me.Controls.Add(Me.Label2)
            Me.Controls.Add(Me.TextBox1)
            Me.Controls.Add(Me.NumericUpDown2)
            Me.Controls.Add(Me.NumericUpDown1)
            Me.FormBorderStyle = FormBorderStyle.Fixed3D
            Me.Icon = DirectCast(manager.GetObject("$this.Icon"), Icon)
            Me.MaximizeBox = False
            Me.Name = "PortScanner"
            Me.StartPosition = FormStartPosition.CenterScreen
            Me.Text = "Port Scanner"
            Me.NumericUpDown1.EndInit
            Me.NumericUpDown2.EndInit
            Me.NumericUpDown3.EndInit
            Me.ResumeLayout(False)
            Me.PerformLayout
        End Sub

        Private Function method_0(ByVal string_0 As String) As IPAddress
            Return Dns.GetHostEntry(string_0).AddressList(0)
        End Function

        Private Function method_1(ByVal ipaddress_0 As IPAddress, ByVal int_0 As Integer, ByVal int_1 As Integer) As Boolean
            Dim flag As Boolean
            Dim socket As New Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp)
            Try 
                socket.BeginConnect(ipaddress_0, int_0, Nothing, 0)
                Dim time As DateTime = DateAndTime.Now.AddMilliseconds(CDbl(int_1))
                Do While (DateTime.Compare(DateAndTime.Now, time) < 0)
                    Thread.Sleep(1)
                    If socket.Connected Then
                        goto Label_0049
                    End If
                Loop
                Return flag
            Label_0049:
                flag = True
            Catch exception1 As Exception
                ProjectData.SetProjectError(exception1)
                ProjectData.ClearProjectError
            Finally
                If (Not socket Is Nothing) Then
                    socket.Dispose
                End If
            End Try
            Return flag
        End Function

        Private Sub method_2(ByVal bool_1 As Boolean)
            Me.NumericUpDown1.Enabled = bool_1
            Me.NumericUpDown2.Enabled = bool_1
            Me.NumericUpDown3.Enabled = bool_1
            Me.Button1.Text = If(bool_1, "Begin", "Cancel")
            Me.bool_0 = Not bool_1
        End Sub

        Private Sub method_3(ByVal object_0 As Object)
            Dim obj2 As Object
            Dim obj3 As Object
            If ForLoopControl.ForLoopInitObj(obj2, NewLateBinding.LateIndexGet(object_0, New Object() { 0 }, Nothing), NewLateBinding.LateIndexGet(object_0, New Object() { 1 }, Nothing), NewLateBinding.LateIndexGet(object_0, New Object() { 2 }, Nothing), obj3, obj2) Then
                Do While Me.bool_0
                    Me.method_4(Me.method_1(DirectCast(NewLateBinding.LateIndexGet(object_0, New Object() { 3 }, Nothing), IPAddress), Conversions.ToInteger(obj2), Conversions.ToInteger(NewLateBinding.LateIndexGet(object_0, New Object() { 4 }, Nothing))), Conversions.ToInteger(obj2))
                    If Not ForLoopControl.ForNextCheckObj(obj2, obj3, obj2) Then
                        Exit Do
                    End If
                Loop
            End If
        End Sub

        Private Sub method_4(ByVal bool_1 As Boolean, ByVal int_0 As Integer)
            If Me.InvokeRequired Then
                Me.Invoke(New OD(AddressOf Me.method_4), New Object() { bool_1, int_0 })
            Else
                If bool_1 Then
                    Me.ListBox1.Items.Add(int_0)
                End If
                Dim bar As ProgressBar = Me.ProgressBar1
                bar.Value += 1
                If (Me.ProgressBar1.Value = Me.ProgressBar1.Maximum) Then
                    Me.method_2(True)
                End If
            End If
        End Sub


        ' Properties
        Friend Overridable Property Button1 As Button
            Get
                Return Me._Button1
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal value As Button)
                Dim handler As EventHandler = New EventHandler(AddressOf Me.Button1_Click)
                If (Not Me._Button1 Is Nothing) Then
                    RemoveHandler Me._Button1.Click, handler
                End If
                Me._Button1 = value
                If (Not Me._Button1 Is Nothing) Then
                    AddHandler Me._Button1.Click, handler
                End If
            End Set
        End Property

        Friend Overridable Property Label2 As Label
            Get
                Return Me._Label2
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal value As Label)
                Me._Label2 = value
            End Set
        End Property

        Friend Overridable Property Label3 As Label
            Get
                Return Me._Label3
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal value As Label)
                Me._Label3 = value
            End Set
        End Property

        Friend Overridable Property Label4 As Label
            Get
                Return Me._Label4
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal value As Label)
                Me._Label4 = value
            End Set
        End Property

        Friend Overridable Property Label5 As Label
            Get
                Return Me._Label5
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal value As Label)
                Me._Label5 = value
            End Set
        End Property

        Friend Overridable Property ListBox1 As ListBox
            Get
                Return Me._ListBox1
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal value As ListBox)
                Me._ListBox1 = value
            End Set
        End Property

        Friend Overridable Property NumericUpDown1 As NumericUpDown
            Get
                Return Me._NumericUpDown1
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal value As NumericUpDown)
                Me._NumericUpDown1 = value
            End Set
        End Property

        Friend Overridable Property NumericUpDown2 As NumericUpDown
            Get
                Return Me._NumericUpDown2
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal value As NumericUpDown)
                Me._NumericUpDown2 = value
            End Set
        End Property

        Friend Overridable Property NumericUpDown3 As NumericUpDown
            Get
                Return Me._NumericUpDown3
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal value As NumericUpDown)
                Me._NumericUpDown3 = value
            End Set
        End Property

        Friend Overridable Property ProgressBar1 As ProgressBar
            Get
                Return Me._ProgressBar1
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal value As ProgressBar)
                Me._ProgressBar1 = value
            End Set
        End Property

        Friend Overridable Property TextBox1 As TextBox
            Get
                Return Me._TextBox1
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal value As TextBox)
                Me._TextBox1 = value
            End Set
        End Property


        ' Fields
        <AccessedThroughProperty("Button1")> _
        Private _Button1 As Button
        <AccessedThroughProperty("Label2")> _
        Private _Label2 As Label
        <AccessedThroughProperty("Label3")> _
        Private _Label3 As Label
        <AccessedThroughProperty("Label4")> _
        Private _Label4 As Label
        <AccessedThroughProperty("Label5")> _
        Private _Label5 As Label
        <AccessedThroughProperty("ListBox1")> _
        Private _ListBox1 As ListBox
        <AccessedThroughProperty("NumericUpDown1")> _
        Private _NumericUpDown1 As NumericUpDown
        <AccessedThroughProperty("NumericUpDown2")> _
        Private _NumericUpDown2 As NumericUpDown
        <AccessedThroughProperty("NumericUpDown3")> _
        Private _NumericUpDown3 As NumericUpDown
        <AccessedThroughProperty("ProgressBar1")> _
        Private _ProgressBar1 As ProgressBar
        <AccessedThroughProperty("TextBox1")> _
        Private _TextBox1 As TextBox
        Private bool_0 As Boolean
        Private icontainer_0 As IContainer

        ' Nested Types
        Public Delegate Sub OD(ByVal open As Boolean, ByVal port As Integer)
    End Class
End Namespace

File: RAT/Hades v1.6/Source/xRAT.cs/xRAT/PortScanner.cs
namespace xRAT
{
    using Microsoft.VisualBasic;
    using Microsoft.VisualBasic.CompilerServices;
    using System;
    using System.ComponentModel;
    using System.Diagnostics;
    using System.Drawing;
    using System.Net;
    using System.Net.Sockets;
    using System.Runtime.CompilerServices;
    using System.Threading;
    using System.Windows.Forms;

    [DesignerGenerated]
    public class PortScanner : Form
    {
        [AccessedThroughProperty("Button1")]
        private Button _Button1;
        [AccessedThroughProperty("Label2")]
        private Label _Label2;
        [AccessedThroughProperty("Label3")]
        private Label _Label3;
        [AccessedThroughProperty("Label4")]
        private Label _Label4;
        [AccessedThroughProperty("Label5")]
        private Label _Label5;
        [AccessedThroughProperty("ListBox1")]
        private ListBox _ListBox1;
        [AccessedThroughProperty("NumericUpDown1")]
        private NumericUpDown _NumericUpDown1;
        [AccessedThroughProperty("NumericUpDown2")]
        private NumericUpDown _NumericUpDown2;
        [AccessedThroughProperty("NumericUpDown3")]
        private NumericUpDown _NumericUpDown3;
        [AccessedThroughProperty("ProgressBar1")]
        private ProgressBar _ProgressBar1;
        [AccessedThroughProperty("TextBox1")]
        private TextBox _TextBox1;
        private bool bool_0;
        private IContainer icontainer_0;

        public PortScanner()
        {
            Class1.QaIGh5M7cuigS();
            this.InitializeComponent();
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            if (this.bool_0)
            {
                this.method_2(true);
            }
            else
            {
                this.method_2(false);
                this.ListBox1.Items.Clear();
                this.ProgressBar1.Maximum = Convert.ToInt32(decimal.Add(this.NumericUpDown2.Value, decimal.One));
                this.ProgressBar1.Minimum = Convert.ToInt32(this.NumericUpDown1.Value);
                this.ProgressBar1.Value = Convert.ToInt32(this.NumericUpDown1.Value);
                try
                {
                    IPAddress address = this.method_0(this.TextBox1.Text);
                    int num = 0;
                    do
                    {
                        new Thread(new ParameterizedThreadStart(this.method_3)).Start(new object[] { decimal.Add(this.NumericUpDown1.Value, new decimal(num)), this.NumericUpDown2.Value, 8, address, this.NumericUpDown3.Value });
                        num++;
                    }
                    while (num <= 7);
                }
                catch (Exception exception1)
                {
                    ProjectData.SetProjectError(exception1);
                    ProjectData.ClearProjectError();
                }
            }
        }

        [DebuggerNonUserCode]
        protected override void Dispose(bool disposing)
        {
            try
            {
                if (disposing && (this.icontainer_0 != null))
                {
                    this.icontainer_0.Dispose();
                }
            }
            finally
            {
                base.Dispose(disposing);
            }
        }

        [DebuggerStepThrough]
        private void InitializeComponent()
        {
            ComponentResourceManager manager = new ComponentResourceManager(typeof(PortScanner));
            this.NumericUpDown1 = new NumericUpDown();
            this.NumericUpDown2 = new NumericUpDown();
            this.TextBox1 = new TextBox();
            this.Label2 = new Label();
            this.Label3 = new Label();
            this.ListBox1 = new ListBox();
            this.NumericUpDown3 = new NumericUpDown();
            this.Label4 = new Label();
            this.Button1 = new Button();
            this.Label5 = new Label();
            this.ProgressBar1 = new ProgressBar();
            this.NumericUpDown1.BeginInit();
            this.NumericUpDown2.BeginInit();
            this.NumericUpDown3.BeginInit();
            this.SuspendLayout();
            this.NumericUpDown1.Font = new Font("Verdana", 8.25f, FontStyle.Regular, GraphicsUnit.Point, 0);
            Point point = new Point(0x80, 70);
            this.NumericUpDown1.Location = point;
            decimal num = new decimal(new int[] { 0xffff, 0, 0, 0 });
            this.NumericUpDown1.Maximum = num;
            this.NumericUpDown1.Name = "NumericUpDown1";
            Size size = new Size(60, 0x15);
            this.NumericUpDown1.Size = size;
            this.NumericUpDown1.TabIndex = 2;
            this.NumericUpDown1.TextAlign = HorizontalAlignment.Center;
            this.NumericUpDown2.Font = new Font("Verdana", 8.25f, FontStyle.Regular, GraphicsUnit.Point, 0);
            point = new Point(0xc2, 70);
            this.NumericUpDown2.Location = point;
            num = new decimal(new int[] { 0xffff, 0, 0, 0 });
            this.NumericUpDown2.Maximum = num;
            this.NumericUpDown2.Name = "NumericUpDown2";
            size = new Size(60, 0x15);
            this.NumericUpDown2.Size = size;
            this.NumericUpDown2.TabIndex = 3;
            this.NumericUpDown2.TextAlign = HorizontalAlignment.Center;
            this.TextBox1.Font = new Font("Verdana", 8.25f, FontStyle.Regular, GraphicsUnit.Point, 0);
            point = new Point(0x80, 0x19);
            this.TextBox1.Location = point;
            this.TextBox1.Name = "TextBox1";
            size = new Size(0x7e, 0x15);
            this.TextBox1.Size = size;
            this.TextBox1.TabIndex = 4;
            this.TextBox1.Text = "127.0.0.1";
            this.TextBox1.TextAlign = HorizontalAlignment.Center;
            this.Label2.AutoSize = true;
            this.Label2.Font = new Font("Verdana", 8.25f, FontStyle.Regular, GraphicsUnit.Point, 0);
            point = new Point(0x7d, 0x36);
            this.Label2.Location = point;
            Padding padding = new Padding(3, 5, 3, 0);
            this.Label2.Margin = padding;
            this.Label2.Name = "Label2";
            size = new Size(70, 13);
            this.Label2.Size = size;
            this.Label2.TabIndex = 5;
            this.Label2.Text = "Port Range";
            this.Label3.AutoSize = true;
            this.Label3.Font = new Font("Verdana", 8.25f, FontStyle.Regular, GraphicsUnit.Point, 0);
            point = new Point(0x7d, 9);
            this.Label3.Location = point;
            this.Label3.Name = "Label3";
            size = new Size(0x2c, 13);
            this.Label3.Size = size;
            this.Label3.TabIndex = 7;
            this.Label3.Text = "Target";
            this.ListBox1.FormattingEnabled = true;
            point = new Point(15, 0x19);
            this.ListBox1.Location = point;
            this.ListBox1.Name = "ListBox1";
            size = new Size(0x6b, 0xba);
            this.ListBox1.Size = size;
            this.ListBox1.TabIndex = 8;
            this.NumericUpDown3.Font = new Font("Verdana", 8.25f, FontStyle.Regular, GraphicsUnit.Point, 0);
            point = new Point(0x80, 0x73);
            this.NumericUpDown3.Location = point;
            num = new decimal(new int[] { 0xea60, 0, 0, 0 });
            this.NumericUpDown3.Maximum = num;
            this.NumericUpDown3.Name = "NumericUpDown3";
            size = new Size(0x7e, 0x15);
            this.NumericUpDown3.Size = size;
            this.NumericUpDown3.TabIndex = 9;
            this.NumericUpDown3.TextAlign = HorizontalAlignment.Center;
            num = new decimal(new int[] { 800, 0, 0, 0 });
            this.NumericUpDown3.Value = num;
            this.Label4.AutoSize = true;
            this.Label4.Font = new Font("Verdana", 8.25f, FontStyle.Regular, GraphicsUnit.Point, 0);
            point = new Point(0x81, 0x63);
            this.Label4.Location = point;
            padding = new Padding(3, 5, 3, 0);
            this.Label4.Margin = padding;
            this.Label4.Name = "Label4";
            size = new Size(0x35, 13);
            this.Label4.Size = size;
            this.Label4.TabIndex = 10;
            this.Label4.Text = "Timeout";
            this.Button1.Font = new Font("Verdana", 8.25f, FontStyle.Regular, GraphicsUnit.Point, 0);
            point = new Point(0x80, 180);
            this.Button1.Location = point;
            this.Button1.Name = "Button1";
            size = new Size(0x7e, 0x1f);
            this.Button1.Size = size;
            this.Button1.TabIndex = 11;
            this.Button1.Text = "Scan Ports";
            this.Button1.UseVisualStyleBackColor = true;
            this.Label5.AutoSize = true;
            this.Label5.Font = new Font("Verdana", 8.25f, FontStyle.Regular, GraphicsUnit.Point, 0);
            point = new Point(12, 9);
            this.Label5.Location = point;
            this.Label5.Name = "Label5";
            size = new Size(70, 13);
            this.Label5.Size = size;
            this.Label5.TabIndex = 12;
            this.Label5.Text = "Open Ports";
            point = new Point(0x80, 0x97);
            this.ProgressBar1.Location = point;
            this.ProgressBar1.Name = "ProgressBar1";
            size = new Size(0x7e, 0x17);
            this.ProgressBar1.Size = size;
            this.ProgressBar1.TabIndex = 13;
            SizeF ef = new SizeF(6f, 13f);
            this.AutoScaleDimensions = ef;
            this.AutoScaleMode = AutoScaleMode.Font;
            size = new Size(0x108, 0xdd);
            this.ClientSize = size;
            this.Controls.Add(this.ProgressBar1);
            this.Controls.Add(this.Label5);
            this.Controls.Add(this.Button1);
            this.Controls.Add(this.Label4);
            this.Controls.Add(this.NumericUpDown3);
            this.Controls.Add(this.ListBox1);
            this.Controls.Add(this.Label3);
            this.Controls.Add(this.Label2);
            this.Controls.Add(this.TextBox1);
            this.Controls.Add(this.NumericUpDown2);
            this.Controls.Add(this.NumericUpDown1);
            this.FormBorderStyle = FormBorderStyle.Fixed3D;
            this.Icon = (Icon) manager.GetObject("$this.Icon");
            this.MaximizeBox = false;
            this.Name = "PortScanner";
            this.StartPosition = FormStartPosition.CenterScreen;
            this.Text = "Port Scanner";
            this.NumericUpDown1.EndInit();
            this.NumericUpDown2.EndInit();
            this.NumericUpDown3.EndInit();
            this.ResumeLayout(false);
            this.PerformLayout();
        }

        private IPAddress method_0(string string_0)
        {
            return Dns.GetHostEntry(string_0).AddressList[0];
        }

        private bool method_1(IPAddress ipaddress_0, int int_0, int int_1)
        {
            bool flag;
            Socket socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
            try
            {
                socket.BeginConnect(ipaddress_0, int_0, null, 0);
                DateTime time = DateAndTime.Now.AddMilliseconds((double) int_1);
                while (DateTime.Compare(DateAndTime.Now, time) < 0)
                {
                    Thread.Sleep(1);
                    if (socket.Connected)
                    {
                        goto Label_0049;
                    }
                }
                return flag;
            Label_0049:
                flag = true;
            }
            catch (Exception exception1)
            {
                ProjectData.SetProjectError(exception1);
                ProjectData.ClearProjectError();
            }
            finally
            {
                if (socket != null)
                {
                    socket.Dispose();
                }
            }
            return flag;
        }

        private void method_2(bool bool_1)
        {
            this.NumericUpDown1.Enabled = bool_1;
            this.NumericUpDown2.Enabled = bool_1;
            this.NumericUpDown3.Enabled = bool_1;
            this.Button1.Text = bool_1 ? "Begin" : "Cancel";
            this.bool_0 = !bool_1;
        }

        private void method_3(object object_0)
        {
            object obj2;
            object obj3;
            if (ObjectFlowControl.ForLoopControl.ForLoopInitObj(obj2, NewLateBinding.LateIndexGet(object_0, new object[] { 0 }, null), NewLateBinding.LateIndexGet(object_0, new object[] { 1 }, null), NewLateBinding.LateIndexGet(object_0, new object[] { 2 }, null), ref obj3, ref obj2))
            {
                while (this.bool_0)
                {
                    this.method_4(this.method_1((IPAddress) NewLateBinding.LateIndexGet(object_0, new object[] { 3 }, null), Conversions.ToInteger(obj2), Conversions.ToInteger(NewLateBinding.LateIndexGet(object_0, new object[] { 4 }, null))), Conversions.ToInteger(obj2));
                    if (!ObjectFlowControl.ForLoopControl.ForNextCheckObj(obj2, obj3, ref obj2))
                    {
                        break;
                    }
                }
            }
        }

        private void method_4(bool bool_1, int int_0)
        {
            if (this.InvokeRequired)
            {
                this.Invoke(new OD(this.method_4), new object[] { bool_1, int_0 });
            }
            else
            {
                if (bool_1)
                {
                    this.ListBox1.Items.Add(int_0);
                }
                ProgressBar bar = this.ProgressBar1;
                bar.Value++;
                if (this.ProgressBar1.Value == this.ProgressBar1.Maximum)
                {
                    this.method_2(true);
                }
            }
        }

        internal virtual Button Button1
        {
            get
            {
                return this._Button1;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                EventHandler handler = new EventHandler(this.Button1_Click);
                if (this._Button1 != null)
                {
                    this._Button1.Click -= handler;
                }
                this._Button1 = value;
                if (this._Button1 != null)
                {
                    this._Button1.Click += handler;
                }
            }
        }

        internal virtual Label Label2
        {
            get
            {
                return this._Label2;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                this._Label2 = value;
            }
        }

        internal virtual Label Label3
        {
            get
            {
                return this._Label3;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                this._Label3 = value;
            }
        }

        internal virtual Label Label4
        {
            get
            {
                return this._Label4;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                this._Label4 = value;
            }
        }

        internal virtual Label Label5
        {
            get
            {
                return this._Label5;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                this._Label5 = value;
            }
        }

        internal virtual ListBox ListBox1
        {
            get
            {
                return this._ListBox1;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                this._ListBox1 = value;
            }
        }

        internal virtual NumericUpDown NumericUpDown1
        {
            get
            {
                return this._NumericUpDown1;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                this._NumericUpDown1 = value;
            }
        }

        internal virtual NumericUpDown NumericUpDown2
        {
            get
            {
                return this._NumericUpDown2;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                this._NumericUpDown2 = value;
            }
        }

        internal virtual NumericUpDown NumericUpDown3
        {
            get
            {
                return this._NumericUpDown3;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                this._NumericUpDown3 = value;
            }
        }

        internal virtual ProgressBar ProgressBar1
        {
            get
            {
                return this._ProgressBar1;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                this._ProgressBar1 = value;
            }
        }

        internal virtual TextBox TextBox1
        {
            get
            {
                return this._TextBox1;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                this._TextBox1 = value;
            }
        }

        public delegate void OD(bool open, int port);
    }
}

